cornice/tests/test_validation.py

```diff
@@ -240,10 +250,17 @@ def test_content_type_with_callable(self):
         response = app.post('/service6', headers={'Content-Type': 'audio/*'},
                             status=415)
         error_description = response.json['errors'][0]['description']
-        self.assertTrue('text/xml' in error_description)
-        self.assertTrue('application/json' in error_description)
+        self.assertIn('text/xml', error_description)
+        self.assertIn('application/json', error_description)
 
-        app.post('/service6', headers={'Content-Type': 'text/xml'})
+    def test_content_type_with_callable_returning_scalar(self):
+        # test that using a callable for content_type works as well
+        # now, the callable returns a scalar instead of a list
+        app = TestApp(main({}))
+        response = app.put('/service6', headers={'Content-Type': 'audio/*'},
+                           status=415)
+        error_description = response.json['errors'][0]['description']
```

Comment:
I'm thinking the same. Would it be fine to define the services required for testing all the nitty-gritty details completely at runtime? When considering this, i can imagine targeting these topics as eventual goals of the test cleanup task:

- The */serviceX* jungle in ``validationapp.py`` would be completely cleaned up, the services will be defined at runtime from the particular test case in ``test_validation.py`` by using a helper function from ``cornice/tests/support.py``, as you suggested. When we're already at spawning the appropriate service at test execution time, let's also revamp the uri namespace to something like */test/service/accept* (former uris */service2* and */service3*) or */test/service/content-type* (former */service5*), to name just two examples to improve on "naming things".
- Just a **single** decorator-based service definition would remain inside the revamped ``validationapp.py`` for testing the decorator machinery against all available http verbs on a single service uri */test/service/decorated*.

Does this sound good to you? As we didn't define services at runtime before and were just using the decorator-style definition in our projects, some hints about doing this would definitively help. Currently, i can't get my head around **not** using Venusian for scanning the service decorator definitions. Thanks in advance!